### PR TITLE
Rework of reward logic

### PR DESCRIPTION
**Pull request by peak3d**
The development of the liquidity pool (LP) staking contract leads to a constant reward per share logic.
Instead providing x% on the asset amount of the staked asset, we now provide a fixed emmision rate for the pool which is distributed over the share holders.
There are 2 benefits: High APR for early investors and calculatable emmissions.

Changes:
[SOL]
- Added Readme.md which describe the deployment steps
- Implement Liquidity Staking Pool, including price extraction via UniV2 Route.
- Re-Implement the stablecoin pool using the same reward logic.
- Finalize / rework controller contract which manages the farms.
- Dummy Booster implementation
- Implement RewardHandler in Token contract, can be moved to an separate contract later.
- Prepare Token contract to signal transactions to external TXWorker contract (for low gas maintanace).

[UI]
- Change store so it can work with private (ganache) network, pass ?network=private to use it.

### CODE REVIEW

File: contracts/src/token/Token.sol [hunk 1]
```diff
@@ -43,51 +49,80 @@ contract WolfToken is ERC20Capped, AccessControl {
...
+  constructor(address _marketingWallet, address _teamWallet)
     ERC20Capped(MAX_SUPPLY)
     ERC20(TOKEN_NAME, TOKEN_SYMBOL)
   {
     // Initialize ERC20 base
     _setupDecimals(TOKEN_DECIMALS);
 
     /*
-     * Mint 9875 into teams wallet
+     * Mint 2375 into teams wallet
```
Comment (juztamau5) [inline contracts/src/token/Token.sol]: `2375` should be `3600` I think

File: contracts/src/investment/UniV2StakeFarm.sol [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+      address(route) != address(0) ? route.getReserves() : (1, 1, 0);
+
+    uint112 swap;
+    // WETH is token 1, swap
+    if ((pairDirection & 1) != 0) {
+      swap = reserve0;
+      reserve0 = reserve1;
+      reserve1 = swap;
+    }
+    // WETH is token 1, swap
```
Comment (juztamau5) [inline contracts/src/investment/UniV2StakeFarm.sol]: Should this be `// WETH is token 2, swap`?

File: contracts/src/crowdsale/Crowdsale.sol [hunk 3]
```diff
@@ -13,14 +13,23 @@ import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
...
+
+import '../../interfaces/uniswap/IUniswapV2Router02.sol';
 
 interface IERC20WolfMintable is IERC20 {
   function mint(address account, uint256 amount) external returns (bool);
 
   function enableUniV2Pair(bool enable) external;
 }
 
+interface IStakeFarm {
```
Comment (juztamau5) [inline contracts/src/crowdsale/Crowdsale.sol]: Can you move the interface to an `interfaces/` directory and add documentation?

File: contracts/src/investment/interfaces/ITxWorker.sol [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright (C) 2020 The Wolfpack
+ * This file is part of wolves.finance - https://github.com/wolvesofwallstreet/wolves.finance
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * See the file LICENSES/README.md for more information.
+ */
+
+pragma solidity >=0.6.0 <0.8.0;
+
+interface ITxWorker {
```
Comment (juztamau5) [inline contracts/src/investment/interfaces/ITxWorker.sol:18]: I like the documentation added to `IFarm.sol`. Not required now but eventually, can you extend the documentation in the other interface files? I just feel like interfaces should be the most clearly documented, as the interfaces are where I initially go to understand the code.

File: contracts/src/investment/UniV2StakeFarm.sol (hunk 2 above)
Comment (peak3d) [reply contracts/src/investment/UniV2StakeFarm.sol]: they have zero based naming, in a uniswap pair you have token0 and token1, I'll remove the space

**Review by juztamau5 — APPROVED** [review]

**Pull request merged**